include/dali/core/host_dev.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_HOST_DEV_H_
+#define DALI_CORE_HOST_DEV_H_
+
+#if defined(__CUDACC__) && !(defined(__CUDA__) && defined(__clang__))
+#define DALI_NO_EXEC_CHECK #pragma nv_exec_check_disable
```

Comment:
out of curiosity: what is this?

Reply:
If there's a piece of `__host__ __device__` code that calls a non-device function, a warning is generated. This warning becomes an error, if the said `__host__ __device__` function is actually used in device code. In templates, however, this is often not the case and this `#pragma` sileces the warning.